packages/babel-plugin/src/index.tsx

```diff
@@ -90,11 +90,17 @@ export default declare<State>((api) => {
             return;
           }
 
-          if (!path.scope.getBinding('React')) {
+          const shouldImportReact = state.opts.importReact === undefined || state.opts.importReact;
+          if (shouldImportReact && !path.scope.getBinding('React')) {
             // React is missing - add it in at the last moment!
             path.unshiftContainer('body', template.ast(`import * as React from 'react'`));
           }
 
+          if (state.compiledImports.styled && !path.scope.getBinding('forwardRef')) {
+            // forwardRef is missing - add it in at the last moment!
+            path.unshiftContainer('body', template.ast(`import { forwardRef } from 'react'`));
```

Comment:
foward ref imported here. its bleeding a bit because this ideally would be in the styled visitor - but that doesnt have any knowledge of the program and i didn't really want to do more work.

we could move it in though.

Reply:
Yes later please

Reply:
created this https://github.com/atlassian-labs/compiled/issues/424